src/main/scala/beam/router/RoutingWorker.scala

```diff
@@ -340,6 +362,30 @@ class RoutingWorker(workerParams: R5Parameters) extends Actor with ActorLogging
       List()
     }
   }
+
+  private def joinResponsesOrCallR5(modesToExclude: List[BeamMode], request: RoutingRequest, responses: Option[RoutingResponse]*) = {
```

Comment:
I spent a bit of time wrapping my head around this only to realize that the original code had a seeming bug in it. The types do not match. In some cases it returns a `RoutingResponse` and in others it returns an `Option[RoutingResponse]`

Here is my stab at stabilizing this method:

```
    if (modesToExclude.isEmpty) {
      r5.calcRoute(request)
    } else {
      val filteredStreetVehicles = request.streetVehicles.filterNot(it => modesToExclude.contains(it.mode))
      val r5ResponseOption = if (filteredStreetVehicles.isEmpty) {
        None
      } else {
        Some(r5.calcRoute(request.copy(streetVehicles = filteredStreetVehicles)))
      }

      val definedResponses = responses.flatten
      definedResponses.headOption.map{
        _.copy(definedResponses.flatMap(_.itineraries).toSeq ++
            r5ResponseOption.map(_.itineraries).getOrElse(Seq.empty))
      }.orElse(r5ResponseOption)
      .getOrElse(/*What to do here - this is the case where there is NO response*/)
```

Note that there is some undefined behavior because it previously returned an Option where the None was not being handled.

Reply:
Justin, thanks for your fix. R5 should be called for full request in the last case.